### PR TITLE
Adds save credit card

### PR DESCRIPTION
**Pull request by ata-n**

### CODE REVIEW

File: traveler_stripe_payment_provider/src/main/java/com/guestlogix/traveler_stripe_payment_provider/TravelerEphemeralKeyProvider.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    public void createEphemeralKey(@NonNull String apiVersion, @NonNull EphemeralKeyUpdateListener keyUpdateListener) {
+        Traveler.fetchEphemeralStripeCustomerKey(apiVersion, new EphemeralKeyFetchCallback() {
+            @Override
+            public void onEphemeralKeyFetchSuccess(EphemeralKey key) {
+                keyUpdateListener.onKeyUpdate(key.getKey());
+            }
+
+            @Override
+            public void onEphemeralKeyError(Error error) {
+                // TODO: Pass responseCode and message from Error when casted as a NetworkError
```
Comment (Alvtag) [inline traveler_stripe_payment_provider/src/main/java/com/guestlogix/traveler_stripe_payment_provider/TravelerEphemeralKeyProvider.java]: Is there a procedure on leaving TODOs in code? e.g. link to tickets, etc

File: traveler_stripe_payment_provider/src/main/java/com/guestlogix/traveler_stripe_payment_provider/StripePaymentManager.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+            @Override
+            public void onError(int errorCode, @NonNull String errorMessage, @Nullable StripeError stripeError) {
+                callback.onPaymentsFetchError(new Error(errorMessage));
+            }
+        });
+    }
+
+    @Override
+    public void savePayment(Payment payment) {
```
Comment (Alvtag) [inline traveler_stripe_payment_provider/src/main/java/com/guestlogix/traveler_stripe_payment_provider/StripePaymentManager.java]: would it make sense for savePayment to take in a CustomerSession.PaymentMethodRetrievalListener()?

File: traveler_ui_kit/src/main/java/com/guestlogix/traveleruikit/fragments/BillingInformationCollectionFragment.java [hunk 3]
```diff
@@ -179,11 +176,21 @@ public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
             holder.label.setText(a.getLabel());
             holder.value.setText(a.getValue());
             holder.delete.setTag(position);
+            holder.itemView.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    selectedIndex = position;
+
+                    adapter.notifyDataSetChanged();
```
Comment (Alvtag) [inline traveler_ui_kit/src/main/java/com/guestlogix/traveleruikit/fragments/BillingInformationCollectionFragment.java]: notifyItemChanged(pos) should work here, would make insertion smoother

hmm, not sure why tagging (`holder.delete.setTag(position);`) is used. we should be able to use the position param of `onBindViewHolder(@NonNull ViewHolder holder, int position) {`

File: traveler_ui_kit/src/main/java/com/guestlogix/traveleruikit/activities/QuestionsActivity.java [hunk 4]
```diff
@@ -85,9 +91,26 @@ protected void onSaveInstanceState(@NonNull Bundle outState) {
...
+            @Override
+            public void onPaymentsFetchSuccess(List<Payment> paymentList) {
+                advanceToSummary(new ArrayList<>(paymentList));
+            }
+
+            @Override
+            public void onPaymentsFetchError(Error error) {
+                // Graceful failure: In case we can't fetch saved payments we still
+                // want to advance the user
+                advanceToSummary(new ArrayList<>());
```
Comment (Alvtag) [inline traveler_ui_kit/src/main/java/com/guestlogix/traveleruikit/activities/QuestionsActivity.java]: Might make sense to make `payments` Nullable so `OrderSummaryActivity` can display an appropriate error message

**Comment by Alvtag** [comment]
QuestionsActivity.java conflict

File: traveler_stripe_payment_provider/src/main/java/com/guestlogix/traveler_stripe_payment_provider/TravelerEphemeralKeyProvider.java (hunk 1 above)
Comment (ata-n) [reply traveler_stripe_payment_provider/src/main/java/com/guestlogix/traveler_stripe_payment_provider/TravelerEphemeralKeyProvider.java]: There should be lol

Comment (ata-n) [reply traveler_stripe_payment_provider/src/main/java/com/guestlogix/traveler_stripe_payment_provider/TravelerEphemeralKeyProvider.java]: There should be, our JIRA barely captures the feature at this time, so I want it to exist somewhere at least and have visibility for the devs

**Pull request closed without merging**